daphne_server/examples/service.rs

```diff
@@ -0,0 +1,82 @@
+use std::path::PathBuf;
+
+use clap::Parser;
+use daphne::DapGlobalConfig;
+use daphne_server::{router, App};
+use daphne_service_utils::Role;
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+struct Config {
+    global: DapGlobalConfig,
+    port: u16,
+    role: Role,
+}
+
+impl TryFrom<Args> for Config {
+    type Error = config::ConfigError;
+    fn try_from(
+        Args {
+            configuration,
+            role,
+            port,
+        }: Args,
+    ) -> Result<Self, Self::Error> {
+        config::Config::builder()
+            .set_default("port", 3000)?
+            .add_source(match configuration {
+                Some(path) => config::File::from(path.as_ref()),
+                None => config::File::with_name("configuration"),
+            })
+            .set_override_option(
+                "role",
+                role.map(|role| config::Value::new(Some(&String::from("args.role")), role)),
+            )?
+            .set_override_option(
+                "port",
+                port.map(|port| config::Value::new(Some(&String::from("args.port")), port)),
+            )?
+            .add_source(config::Environment::with_prefix("DAP"))
+            .build()?
+            .try_deserialize()
+    }
+}
+
+#[derive(clap::Parser)]
+struct Args {
+    #[arg(short, long)]
```

Comment:
CLI documentation should indicate that a toml file is expected.

Reply:
that's actually not true :smirk: you can use any kind of file as long as it's called `configuration` :eyes: